pkg/scalers/azure_queue_scaler.go

```diff
@@ -91,105 +87,63 @@ func NewAzureQueueScaler(config *scalersconfig.ScalerConfig) (Scaler, error) {
 	}, nil
 }
 
-func parseAzureQueueMetadata(config *scalersconfig.ScalerConfig, logger logr.Logger) (*azureQueueMetadata, kedav1alpha1.AuthPodIdentity, error) {
+func parseAzureQueueMetadata(config *scalersconfig.ScalerConfig) (azureQueueMetadata, kedav1alpha1.AuthPodIdentity, error) {
 	meta := azureQueueMetadata{}
-	meta.targetQueueLength = defaultTargetQueueLength
-
-	if val, ok := config.TriggerMetadata[queueLengthMetricName]; ok {
-		queueLength, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			logger.Error(err, "Error parsing azure queue metadata", "queueLengthMetricName", queueLengthMetricName)
-			return nil, kedav1alpha1.AuthPodIdentity{},
-				fmt.Errorf("error parsing azure queue metadata %s: %w", queueLengthMetricName, err)
-		}
-
-		meta.targetQueueLength = queueLength
+	err := config.TypedConfig(&meta)
+	if err != nil {
+		return meta, kedav1alpha1.AuthPodIdentity{}, fmt.Errorf("error parsing azure queue metadata: %w", err)
 	}
 
-	meta.activationTargetQueueLength = 0
-	if val, ok := config.TriggerMetadata[activationQueueLengthMetricName]; ok {
-		activationQueueLength, err := strconv.ParseInt(val, 10, 64)
-		if err != nil {
-			logger.Error(err, "Error parsing azure queue metadata", activationQueueLengthMetricName, activationQueueLengthMetricName)
-			return nil, kedav1alpha1.AuthPodIdentity{},
-				fmt.Errorf("error parsing azure queue metadata %s: %w", activationQueueLengthMetricName, err)
-		}
-
-		meta.activationTargetQueueLength = activationQueueLength
+	err = meta.Validate()
```

Comment:
Validate() is not needed to be called. It will be called during TypedConfig().

Reply:
solved